### ISSUE TITLE
Linting a binary files results in an exception

### ISSUE THREAD

**Issue by cmmeyer** [awslabs/cfn-python-lint#357]
*cfn-lint version: (`cfn-lint --version`)*
0.7.2

*Description of issue.*
Attempting to lint a binary files results in an exception:

Please provide as much information as possible:
* Template linting issues:

`cfn-lint CloudFormationEndpointSignals.zip` resulted in:

```
Traceback (most recent call last):
  File "/usr/local/bin/cfn-lint", line 11, in <module>
    load_entry_point('cfn-lint==0.7.2', 'console_scripts', 'cfn-lint')()
  File "/usr/local/lib/python3.6/site-packages/cfn_lint-0.7.2-py3.6.egg/cfnlint/__main__.py", line 30, in main
    (template, rules, template_matches) = cfnlint.core.get_template_rules(filename, args)
  File "/usr/local/lib/python3.6/site-packages/cfn_lint-0.7.2-py3.6.egg/cfnlint/core.py", line 245, in get_template_rules
    (template, matches) = cfnlint.decode.decode(filename, args.ignore_bad_template)
  File "/usr/local/lib/python3.6/site-packages/cfn_lint-0.7.2-py3.6.egg/cfnlint/decode/__init__.py", line 40, in decode
    template = cfnlint.decode.cfn_yaml.load(filename)
  File "/usr/local/lib/python3.6/site-packages/cfn_lint-0.7.2-py3.6.egg/cfnlint/decode/cfn_yaml.py", line 206, in load
    return loads(fp.read(), filename)
  File "/usr/local/lib/python3.6/site-packages/cfn_lint-0.7.2-py3.6.egg/cfnlint/decode/cfn_yaml.py", line 191, in loads
    loader = MarkedLoader(yaml_string, fname)
  File "/usr/local/lib/python3.6/site-packages/cfn_lint-0.7.2-py3.6.egg/cfnlint/decode/cfn_yaml.py", line 140, in __init__
    Reader.__init__(self, stream)
  File "/usr/local/lib/python3.6/site-packages/yaml/reader.py", line 74, in __init__
    self.check_printable(stream)
  File "/usr/local/lib/python3.6/site-packages/yaml/reader.py", line 144, in check_printable
    'unicode', "special characters are not allowed")
yaml.reader.ReaderError: unacceptable character #x0000: special characters are not allowed
  in "<unicode string>", position 0
```

**Comment by cmmeyer** [comment]
```
cfn-lint CloudFormationEndpointSignals.zip
2018-10-26 00:51:40,695 - cfnlint.decode - ERROR - Cannot read file contents: CloudFormationEndpointSignals.zip
E0000 Cannot read file contents: CloudFormationEndpointSignals.zip
CloudFormationEndpointSignals.zip:1:1
```
So clean!

**Comment by cmmeyer** [comment]
Resolved with #373
